### PR TITLE
[K8s-array] Add default annotations/labels

### PR DESCRIPTION
**Pull request by akhurana001**
# TL;DR
Since K8s-array pods don't go through common resource-creation logic in Propeller, labels/annotations are missed. Adding them similar to how we do for spark plugin.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by anandswaminathan — DISMISSED** [review]

**Review by anandswaminathan — APPROVED** [review]

File: go/tasks/plugins/array/k8s/transformer.go [hunk 1]
```diff
@@ -81,6 +84,9 @@ func FlyteArrayJobToK8sPodTemplate(ctx context.Context, tCtx core.TaskExecutionC
 		return v1.Pod{}, nil, err
 	}
 
+	annotations := utils.UnionMaps(config.GetK8sPluginConfig().DefaultAnnotations, tCtx.TaskExecutionMetadata().GetAnnotations())
```
Comment (milton0825) [inline go/tasks/plugins/array/k8s/transformer.go:87]: So `tCtx.TaskExecutionMetadata().GetAnnotations()` is always going to be empty by design?

Comment (anandswaminathan) [reply go/tasks/plugins/array/k8s/transformer.go:87]: tCtx.TaskExecutionMetadata().GetAnnotations() comes from the workflow/task. On top we add few default ones that are set in Propeller config.

Comment (akhurana001) [reply go/tasks/plugins/array/k8s/transformer.go:87]: yes, currently we miss the default labels/annotations  so adding those in this PR.

**Comment by milton0825** [comment]
👍
